ci-operator/step-registry/rosa/cluster/wait-ready/operators/rosa-cluster-wait-ready-operators-commands.sh

```diff
@@ -66,13 +68,16 @@ else
   echo "All cluster operators ready after $(( ${end_time} - ${start_time} )) seconds"
 fi
 
-# Additionally, OSD has a job "osd-cluster-ready" that determines when
-# the managed service operators are done and the cluster is truly ready.
-# Don't consider the cluster ready until that's done.
-while ! oc get job/osd-cluster-ready -n openshift-monitoring &>/dev/null; do
-  echo "Waiting for job/osd-cluster-ready to exist..."
-  sleep 30
-done
+if [[ "$HOSTED_CP" == "false" ]]; then
+  # Additionally, OSD has a job "osd-cluster-ready" that determines when
+  # the managed service operators are done and the cluster is truly ready.
+  # Don't consider the cluster ready until that's done.
+  # Only check for ROSA Classic clusters
+  while ! oc get job/osd-cluster-ready -n openshift-monitoring &>/dev/null; do
+    echo "Waiting for job/osd-cluster-ready to exist..."
+    sleep 30
+  done
 
-echo "Waiting for job/osd-cluster-ready to complete..."
-oc wait --for=condition=complete job/osd-cluster-ready -n openshift-monitoring --timeout=60m
+  echo "Waiting for job/osd-cluster-ready to complete..."
+  oc wait --for=condition=complete job/osd-cluster-ready -n openshift-monitoring --timeout=60m
```

Comment:
The `while` block and `oc wait` are duplicate actions. I'd prefer the `oc wait` with the 60m timeout.

Besides, for your changes also affect the ROSA classic, it is better to trigger a ROSA classic job too.

Reply:
Yes, it makes sense to do a existing checking first. But how long the waiting for the job exist could be? How long the waiting time for the job to be complete be? 

As I know, if there is no timeout defined in the ref.yaml,  the default timeout for a step(pod) is 2h. So I advice:
1. let us give a timeout to terminate the `while` loop.
2. I think the job should be started and completed in 1h(I guess it may be a user experience bug if the time is too long), so let us sum the `oc wait` + `while` < 60m.  Otherwise, we need to config timeout in the ref.yaml.